tests/e2e/common/common.go

```diff
@@ -386,6 +398,78 @@ func CRDTest(details VersionDetails, opts TestOptions) func(t *testing.T) {
 	}
 }
 
+func GenerateNewCertAndRenew(details VersionDetails, opts TestOptions) func(t *testing.T) {
+	return func(t *testing.T) {
+		daprPath := getDaprPath()
+		err := exportCurrentCertificate(daprPath)
+		require.NoError(t, err, "expected no error on certificate exporting")
+
+		output, err := spawn.Command(daprPath, "mtls", "renew-certificate", "-k", "--valid-until", "20", "--restart")
+		t.Log(output)
+		require.NoError(t, err, "expected no error on certificate renewal")
+
+		done := make(chan struct{})
+		podsRunning := make(chan struct{})
+
+		go waitAllPodsRunning(t, DaprTestNamespace, opts.HAEnabled, done, podsRunning)
+		select {
+		case <-podsRunning:
+			t.Logf("verified all pods running in namespace %s are running after certficate change", DaprTestNamespace)
+		case <-time.After(2 * time.Minute):
+			done <- struct{}{}
+			t.Logf("timeout verifying all pods running in namespace %s", DaprTestNamespace)
+			t.FailNow()
+		}
+		assert.Contains(t, output, "Certificate rotation is successful!")
+	}
+}
+
+func UseProvidedNewCertAndRenew(details VersionDetails) func(t *testing.T) {
+	return func(t *testing.T) {
+		daprPath := getDaprPath()
+		args := []string{
+			"mtls", "renew-certificate", "-k",
+			"--ca-root-certificate", "./certs/ca.crt",
+			"--issuer-private-key", "./certs/issuer.key",
+			"--issuer-public-certificate", "./certs/issuer.crt",
+			"--restart",
+		}
+		output, err := spawn.Command(daprPath, args...)
+		t.Log(output)
+		require.NoError(t, err, "expected no error on certificate renewal")
+		assert.Contains(t, output, "Certificate rotation is successful!")
+
+		// remove cert directory created earlier.
+		os.RemoveAll("./certs")
```

Comment:
not like to have a folder created by one test removed by another .... can this be in the Kubernetes_test.go file instead of common ?
A single flow ... 
1. create a folder `./certs`
2. Run GenerateNewCerts -- export cert to the previously created folder 
3. Run the UserProvidedNewCertsAndRenew test ....
4. Remove the ./certs folder that was created in step 1

1 and 4 in the Test function ... and the rest can be what it is now ...